### PR TITLE
fix: support adding a single instruction to moments

### PR DESCRIPTION
**Pull request by guomanmin**
*Description of changes:*

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-sdk-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by math411** [comment]
Running `tox` will take care of the formatting issues and stage those for a new commit

File: src/braket/circuits/moments.py [hunk 1]
```diff
@@ -163,17 +164,21 @@ def time_slices(self) -> Dict[int, List[Instruction]]:
...
         Args:
-            instructions (Iterable[Instruction]): Instructions to add to self. The instruction is
-                added to the max time slice in which the instruction fits.
+            instructions (Union[Iterable[Instruction], Instruction]): Instructions to add to self.
+            The instruction is added to the max time slice in which the instruction fits.
             noise_index (int): the number of noise channels at the same moment. For gates, this
-                is the number of gate_noise channels associated with that gate. For all other noise
-                types, noise_index starts from 0; but for gate noise, it starts from 1.
+            is the number of gate_noise channels associated with that gate. For all other noise
+            types, noise_index starts from 0; but for gate noise, it starts from 1.
```
Comment (speller26) [inline src/braket/circuits/moments.py:178]: Keep the indentation the way it was before; the docs need that formatting

File: src/braket/circuits/moments.py [hunk 2]
```diff
@@ -163,17 +164,21 @@ def time_slices(self) -> Dict[int, List[Instruction]]:
...
+    ) -> None:
         """
-        Add instructions to self.
+        Add one or more instructions to self.
 
         Args:
-            instructions (Iterable[Instruction]): Instructions to add to self. The instruction is
-                added to the max time slice in which the instruction fits.
+            instructions (Union[Iterable[Instruction], Instruction]): Instructions to add to self.
+            The instruction is added to the max time slice in which the instruction fits.
```
Comment (speller26) [inline src/braket/circuits/moments.py:175]:
```suggestion
                The instruction is added to the max time slice in which the instruction fits.
```

File: src/braket/circuits/moments.py (hunk 1 above)
Comment (speller26) [inline src/braket/circuits/moments.py:178]:
```suggestion
                is the number of gate_noise channels associated with that gate. For all other noise
                types, noise_index starts from 0; but for gate noise, it starts from 1.
```

**Review by speller26 — APPROVED** [review]

**Pull request merged**